src/content/app/genome-browser/components/track-panel/components/track-panel-list/track-panel-items/TrackPanelRegulationLegend.tsx

```diff
@@ -42,6 +43,26 @@ const TrackPanelRegulationLegend = (props: { disabled: boolean }) => {
   const dispatch = useDispatch();
   const { trackDrawerOpened, reportTrackPanelSectionToggled } =
     useGenomeBrowserAnalytics();
+  const { data } = useGetSpeciesStatisticsQuery(
+    {
+      genomeId: activeGenomeId ?? ''
+    },
+    {
+      skip: !activeGenomeId
+    }
+  );
+
+  const regulationStats = data?.genome_stats.regulation_stats;
+
+  if (!regulationStats) {
+    return;
+  }
+
+  const regulationStatKeys = Object.keys(regulationStats);
+  const availableRegulatoryFeatures = regulationLegend.filter((legend) => {
+    return regulationStatKeys.includes(legend.id);
+  });
```

Comment:
The check above is incorrect. The statistics object coming from the backend will always include all keys; but the values of some of those keys will be null

See Chicken. All regulatory statistics fields are there; but some are marked as null. Notice that the legend in the sidebar is listing all features:

<img width="1439" alt="image" src="https://github.com/Ensembl/ensembl-client/assets/6834224/dd73376d-c1d1-4e60-ac62-1343bd89efe0">

Change

```
regulationLegend.filter((legend) => {
  return regulationStatKeys.includes(legend.id);
});
```
to

```
regulationLegend.filter((item) => {
  return !!regulationStatKeys[item];
});
```

Reply:
this doesn't work
![Screenshot 2023-11-20 at 15 06 54](https://github.com/Ensembl/ensembl-client/assets/6347854/1f4fea16-4097-4b92-abe1-9acd89b8f4e9)

Reply:
Sorry, not `!!regulationStatKeys[item]`, but `!!regulationStatKeys[item.id]`. An `item` is an object, and, of course, cannot be used as a key. My mistake.